analysis/study_definition.py

```diff
@@ -313,34 +456,97 @@
         include_date_of_match = True,
         date_format = "YYYY-MM-DD",
     ),
-    ## Reduced kidney function
-    creatinine = patients.with_these_clinical_events(
-        creatinine_codes, # imported from codelists.py
+    ### Dialysis
+    dialysis = patients.with_these_clinical_events(
+        dialysis_codes, # imported from codelists.py
+        returning = "binary_flag",
+        on_or_before = "index_date",
+        find_last_match_in_period = True,
+    ),
+    #### Date of dialysis
+    dialysis_date = patients.with_these_clinical_events(
+        dialysis_codes, # imported from codelists.py
+        returning = "date",
+        on_or_before = "index_date",
+        find_last_match_in_period = True,
+        date_format = "YYYY-MM-DD",
+    ),
+    ### Kidney transplant 
+    kidney_transplant = patients.with_these_clinical_events(
+        kidney_transplant_codes,
+        returning = "binary_flag",
+        on_or_before = "index_date",
+        find_last_match_in_period = True,        
+    ),
+    #### Date of kidney transplant
+    kidney_transplant_date = patients.with_these_clinical_events(
+        kidney_transplant_codes,
+        returning = "date",
+        on_or_before = "index_date",
+        find_last_match_in_period = True, 
+        date_format = "YYYY-MM-DD",       
+    ),
+    #### Categorise dialysis in dialysis with previous kidney transplant; dialysis without previous transplant
+    dialysis_kidney_transplant = patients.categorised_as(
+        {
+            "0": "DEFAULT",
+            "1": """
+                (dialysis AND kidney_transplant) AND kidney_transplant_date <= dialysis_date
+            """,
+            "2": """
+                (dialysis AND NOT kidney_transplant) OR ((dialysis AND kidney_transplant) AND kidney_transplant_date > dialysis_date)
+            """,
+        },
+        return_expectations = {"category": {"ratios": {"0": 0.8, "1": 0.1, "2": 0.1}},},
+    ),
+    ### eGFR 
+    #### egfr_flag is needed because missing egfr values will be coded as 0, and we need to make a 
+    #### distinction between missing and not missing in variable 'ckd' below
+    egfr_flag = patients.with_these_clinical_events(
+        egfr_codes, # imported from codelists.py
+        returning = "binary_flag",
+        find_last_match_in_period = True,
+        return_expectations = {
+            "incidence": 0.95,
+        },
+    ),
+    egfr = patients.with_these_clinical_events(
```

Comment:
As with hba1c, may want to investigate the operator here

Reply:
Thx! I've incorporated the comparator in the definition of the vars.